src/twisted/mail/pop3client.py

```diff
@@ -1,1238 +1,24 @@
-            L{ServerErrorResponse}
-        @return: A deferred which fires when the server response is received.
-            On an OK response, the deferred succeeds with the number of
-            messages in the mailbox and the size of the mailbox in octets.
-            On an ERR response, the deferred fails with a server error
-            response failure.
-        """
-        return self.sendShort(b"STAT", None).addCallback(_statXform)
-
-    def listSize(self, consumer=None):
-        """
-        Send a LIST command to retrieve the sizes of all messages on the
-        server.
-
-        @type consumer: L{None} or callable that takes
-            2-L{tuple} of (0) L{int}, (1) L{int}
-        @param consumer: A function which consumes the 0-based message index
-            and message size derived from the server response.
-
-        @rtype: L{Deferred <defer.Deferred>} which fires L{list} of L{int} or
-            callable that takes 2-L{tuple} of (0) L{int}, (1) L{int}
-        @return: A deferred which fires when the entire response has been
-            received.  When a consumer is not provided, the return value is a
-            list of message sizes.  Otherwise, it returns the consumer itself.
-        """
-        return self._consumeOrSetItem(b"LIST", None, consumer, _listXform)
-
-    def listUID(self, consumer=None):
-        """
-        Send a UIDL command to retrieve the UIDs of all messages on the server.
-
-        @type consumer: L{None} or callable that takes
-            2-L{tuple} of (0) L{int}, (1) L{bytes}
-        @param consumer: A function which consumes the 0-based message index
-            and UID derived from the server response.
-
-        @rtype: L{Deferred <defer.Deferred>} which fires with L{list} of
-            L{object} or callable that takes 2-L{tuple} of (0) L{int},
-            (1) L{bytes}
-        @return: A deferred which fires when the entire response has been
-            received.  When a consumer is not provided, the return value is a
-            list of message sizes.  Otherwise, it returns the consumer itself.
-        """
-        return self._consumeOrSetItem(b"UIDL", None, consumer, _uidXform)
-
-    def quit(self):
-        """
-        Send a QUIT command to disconnect from the server.
-
-        @rtype: L{Deferred <defer.Deferred>} which successfully fires with
-            L{bytes} or fails with L{ServerErrorResponse}
-        @return: A deferred which fires when the server response is received.
-            On an OK response, the deferred succeeds with the server
-            response minus the status indicator.  On an ERR response, the
-            deferred fails with a server error response failure.
-        """
-        return self.sendShort(b"QUIT", None)
-
+warnings.warn(
+    "twisted.mail.pop3client was deprecated in Twisted NEXT. Use twisted.mail.pop3 instead.",
+    DeprecationWarning,
+    stacklevel=2,
+)
 
-__all__ = []  # type: List[str]
```

Comment:
It seems very unlikely that someone would have `from twisted.mail.pop3client import *` in their code, but if that line does somehow exist somewhere, keeping `__all__ = []` would be better, since it preserves the old situation.

Reply:
I have added an empty all.